scrapy/downloadermiddlewares/httpproxy.py

```diff
@@ -26,7 +26,7 @@ def _get_proxy(self, url, orig_type):
         proxy_url = urlunparse((proxy_type or orig_type, hostport, '', '', '', ''))
 
         if user:
-            user_pass = '%s:%s' % (unquote(user), unquote(password))
+            user_pass = to_bytes('%s:%s' % (unquote(user), unquote(password)))
```

Comment:
This uses utf8 encoding; is it a right thing to do? HTTP Basic Auth usually uses latin1, but I don't know about Proxy-Autorization header. Maybe it makes sense to use https://github.com/scrapy/w3lib/blob/7cd5e23f0d790ba91dd208ec7043c65130dcd104/w3lib/http.py#L71 to have a single code path.